src/test/rustdoc-ui/automatic-links.rs

```diff
@@ -0,0 +1,22 @@
+#![deny(automatic_links)]
+
+/// [http://a.com](http://a.com)
+//~^ ERROR unneeded long form for URL
+/// [http://b.com]
+//~^ ERROR unneeded long form for URL
+///
+/// [http://b.com]: http://b.com
+///
+/// [http://c.com][http://c.com]
+pub fn a() {}
+
+/// https://somewhere.com?hello=12
+//~^ ERROR won't be a link as is
```

Comment:
Can you also test URL fragments?
```suggestion
/// https://somewhere.com?hello=12
//~^ ERROR won't be a link as is
/// https://example.com?hello=12#xyz
//~^ ERROR
/// https://example.com#xyz
//~^ ERROR
```